### PR TITLE
Ignore -ivfsoverlay parameter

### PR DESCRIPTION
**Pull request by niw**
**Problems**

When using clang with Xcode, there is an error, ccache detects multiple source files as input:
```
[2020-10-22T13:28:08.137445 43404] Hostname: steipete-mbp-i9.local
[2020-10-22T13:28:08.137484 43404] Working directory: /Users/steipete/Projects/PSPDFKit/iOS/PSPDFKit
[2020-10-22T13:28:08.137910 43404] Set profile directory to .
[2020-10-22T13:28:08.137983 43404] /Users/steipete/Builds/PSPDFKit-apwojecgwxmuoieabhwzqzmecixq/Index/DataStore is not a regular file, not considering as input file
[2020-10-22T13:28:08.138633 43404] Multiple input files: /Users/steipete/Builds/PSPDFKit-apwojecgwxmuoieabhwzqzmecixq/Build/Intermediates.noindex/PSPDFKit.build/Debug-iphonesimulator/PSPDFKit.framework.build/all-product-headers.yaml and /Users/steipete/Projects/PSPDFKit/iOS/PSPDFKit/Glyphs/PSPDFTextBlock.m
[2020-10-22T13:28:08.138868 43404] Failed; falling back to running the real compiler
[2020-10-22T13:28:08.139051 43404] Executing /usr/bin/clang -x objective-c -target x86_64-apple-ios12.0-simulator -fmessage-length=0 -fdiagnostics-show-note-include-stack (...) -ivfsoverlay /Users/steipete/Builds/PSPDFKit-apwojecgwxmuoieabhwzqzmecixq/Build/Intermediates.noindex/PSPDFKit.build/Debug-iphonesimulator/PSPDFKit.framework.build/all-product-headers.yaml -MMD -MT dependencies -MF /Users/steipete/Builds/PSPDFKit-apwojecgwxmuoieabhwzqzmecixq/Build/Intermediates.noindex/PSPDFKit.build/Debug-iphonesimulator/PSPDFKit.framework.build/Objects-normal/x86_64/PSPDFTextBlock.d --serialize-diagnostics /Users/steipete/Builds/PSPDFKit-apwojecgwxmuoieabhwzqzmecixq/Build/Intermediates.noindex/PSPDFKit.build/Debug-iphonesimulator/PSPDFKit.framework.build/Objects-normal/x86_64/PSPDFTextBlock.dia -c /Users/steipete/Projects/PSPDFKit/iOS/PSPDFKit/Glyphs/PSPDFTextBlock.m -o /Users/steipete/Builds/PSPDFKit-apwojecgwxmuoieabhwzqzmecixq/Build/Intermediates.noindex/PSPDFKit.build/Debug-iphonesimulator/PSPDFKit.framework.build/Objects-normal/x86_64/PSPDFTextBlock.o
[2020-10-22T13:28:08.139143 43404] Result: multiple source files
```
(Removed most parameters that are not relevant for this to improve readability)

The `all-product-headers.yaml` file contains following content:
```
{"case-sensitive":"false","roots":[{"contents":[{"external-contents":"/Users/steipete/Projects/PSPDFKit/iOS/PSPDFKit/Encryption/PSPDFAESCryptoDataProvider.h","name":"PSPDFAESCryptoDataProvider.h","type":"file"},{"external-contents":"/Users/steipete/Projects/PSPDFKit/iOS/PSPDFKit/Encryption/PSPDFAESCryptoDataSink.h","name":"PSPDFAESCryptoDataSink.h","type":"file"},{"exte
```
The `-ivfsoverlay` feature seems to be used when combining Objective-C and Swift code, to provide a virtual file system overlay. It is not 100% certain in what way this affects caching, but ignoring the property works great and and produces correct results with cached content.

**Solution**

Ignores clang `-ivfsoverlay <arg>` parameter that is often used by Xcode
with an optional new `ivfsoverlay` sloppiness parameter.

This pull request and patch is based on original pull request and patch #700.

### CODE REVIEW

**Comment by niw** [comment]
@steipete I added your name as author in the pull request description which will be a part of commit message because git doesn't have ability to have multiple author in a single commit. However, if you prefer to take the author of the commit, I will rewrite author of commit. Please let me know the best approach for you!

**Comment by steipete** [comment]
Yes, I'd like to have the contributor badge and author on this. You can take my changes and then add another commit with your changes on it, so it's correctly represented.

**Comment by steipete** [comment]
From what I understood in https://github.com/ccache/ccache/pull/700 is that there are tests needed to land this - this was my main blocker. Updating the PR to be working on master again is the easy part.

**Comment by niw** [comment]
@steipete okay, I restore you commits in #700 which I was squashed initially.
And yes, test is not added yet, also some CI failed somehow.

**Comment by steipete** [comment]
Thank you! 🙏

**Comment by niw** [comment]
So these CI tests failures are caused by `libzstd1` dependencies issue, not by this pull request changes.

File: doc/MANUAL.adoc [hunk 1]
```diff
@@ -824,6 +824,10 @@ still has to do _some_ preprocessing (like macros).
     mtime is too new. This sloppiness disables that check. See also
     _<<_handling_of_newly_created_header_files,Handling of newly created header
     files>>_.
+*ivsfsoverlay*::
+    Ignore the Clang compiler option **-ivfsoverlay* and its argument. This is
```
Comment (jrosdahl) [inline doc/MANUAL.adoc:828]:
```suggestion
    Ignore the Clang compiler option *-ivfsoverlay* and its argument. This is
```

**Comment by jrosdahl** [comment]
Great, thanks!

> So these CI tests failures are caused by libzstd1 dependencies issue, not by this pull request changes.

Fixed in aa7fd27609be.

**Comment by steipete** [comment]
Thanks again for picking this up and bringing it to the finish line! I'd love to put ccache on our Open Source sponsor list for GitHub - or one of the core contributors. Let me know if any of that is planned. 🙏

**Pull request merged**